### ISSUE TITLE
Unintended text selection when swiping between tabs

### ISSUE THREAD

**Issue by cdlarson91** [software-mansion/react-native-gesture-handler#1372]
## Description
There is an issue with unintended text selection when swiping between tabs. **I submitted this to the react-navigation repo (https://github.com/react-navigation/react-navigation/issues/9337) and they asked that I include it over here too, because the issue seems to be related to react-native-gesture-handler, which is one of their dependencies.**

## Code
```javascript
import 'react-native-gesture-handler';
import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createMaterialTopTabNavigator } from "@react-navigation/material-top-tabs";

const Tab = createMaterialTopTabNavigator();

export default function App() {
    return (
    <NavigationContainer>
      <Tab.Navigator 
        swipeEnabled={true}>
        <Tab.Screen name="A" component={AScreen} />
        <Tab.Screen name="B" component={BScreen} />
      </Tab.Navigator>
    </NavigationContainer>
  );
}

function AScreen() {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text selectable={true}>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In non est eget elit pretium elementum. Praesent at metus elementum, auctor turpis ac, tristique ex. Praesent malesuada aliquam lacus, nec vestibulum lorem venenatis sed. Nullam tincidunt tristique massa, quis molestie metus. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce arcu ex, pulvinar eu aliquet sed, facilisis et arcu. Proin sit amet mi est.</Text>
    </View>
  );
}

function BScreen() {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text selectable={true}>Sed leo lacus, bibendum cursus dignissim a, suscipit eget nisi. Aliquam vestibulum eros eget sem feugiat tincidunt placerat vel lorem. Pellentesque odio ligula, elementum ullamcorper urna at, cursus pharetra lectus. Aenean laoreet egestas sapien quis congue. Donec dictum, ex at congue faucibus, orci tortor pharetra libero, in laoreet velit enim eu velit. Aliquam porta nunc a lobortis facilisis. Curabitur tincidunt mollis nisl, scelerisque fermentum diam ultrices et. Phasellus eros sapien, mollis nec sodales at, sodales sit amet ante.</Text>
    </View>
  );
}
```

## Steps To Reproduce

It's easier to observe on a physical part because of the swipe action. In short, swipe from Tab A to Tab B by starting the swipe over the lorem ipsum text. Once the swipe is complete, swiping back from Tab B to Tab A without swiping over the text (i.e. starting the swipe either above or below the text) will show that text on Tab A had been selected.

I experienced this is my own project and was able to reproduce it with this very simple snack code below.

### Expected behavior
When swiping between tabs, the swipe should occur without the text being selected, even if the swipe is initiated over the text.

### Actual behavior

When using the Text component with the Text prop selectable={true} on screens under the Material Top Tab Navigator, swiping between the screens causes the text to be selected if the swipe is initiated over the text. This is undesirable, because no matter how fast the swipe is or how short in time the finger is touching the screen, the text is unintentionally highlighted/selected on swipe. The highlight seems to occur during/after the swipe is complete. I've provided sample code above wherein the problem arises.

### Some things worth noting:

- When changing the Tab.Navigator prop swipeEnabled to false and you try to swipe, the issue does not occur, so it appears that its the allowed swipe gesture that's causing the issue.
- **I found an identical issue when scrolling vertically on screens (not on the tab navigator) that used Scrollview imported from 'react-native-gesture-handler'. As soon as I imported the Scrollview from 'react-native' instead the issue went away. I know that 'react-navigation' makes use of 'react-native-gesture-handler' so maybe this issue is related to 'react-native-gesture-handler'?**

## Snack or minimal code example
https://snack.expo.io/@cdlarson91/tab-swipe-selectable-bug

## Package versions

- React:16.13.1
- React Native: 0.63.2
- React Native Gesture Handler:1.7.0

- react-native-screens: ~2.10.1
- react-native-tab-view: ^2.0.0
- react-native-reanimated: ~1.13.0
- react-navigation/native: 5.9.2
- react-native-safe-area-context: 3.1.4
- react-native-community/masked-view: 0.1.10
- react-navigation/material-top-tabs: 5.3.13

**Comment by cdlarson91** [comment]
Unfortunately I am still experiencing this problem. Copyable text is the #1
request now on my app, and I can't provide it without a workaround/fix to
this swiping issue.

**Comment by j-piasecki** [comment]
Hi! Could you try wrapping your selectable text views with `NativeViewGestureHandler`? I believe that it works correctly after this change:
```jsx
import { NativeViewGestureHandler } from 'react-native-gesture-handler';
import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createMaterialTopTabNavigator } from "@react-navigation/material-top-tabs";

const Tab = createMaterialTopTabNavigator();

export default function App() {
    return (
    <NavigationContainer>
      <Tab.Navigator 
        swipeEnabled={true}>
        <Tab.Screen name="A" component={AScreen} />
        <Tab.Screen name="B" component={BScreen} />
      </Tab.Navigator>
    </NavigationContainer>
  );
}

function AScreen() {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <NativeViewGestureHandler>
        <Text selectable>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In non est eget elit pretium elementum. Praesent at metus elementum, auctor turpis ac, tristique ex. Praesent malesuada aliquam lacus, nec vestibulum lorem venenatis sed. Nullam tincidunt tristique massa, quis molestie metus. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce arcu ex, pulvinar eu aliquet sed, facilisis et arcu. Proin sit amet mi est.</Text>
      </NativeViewGestureHandler>
    </View>
  );
}

function BScreen() {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <NativeViewGestureHandler>
        <Text selectable>Sed leo lacus, bibendum cursus dignissim a, suscipit eget nisi. Aliquam vestibulum eros eget sem feugiat tincidunt placerat vel lorem. Pellentesque odio ligula, elementum ullamcorper urna at, cursus pharetra lectus. Aenean laoreet egestas sapien quis congue. Donec dictum, ex at congue faucibus, orci tortor pharetra libero, in laoreet velit enim eu velit. Aliquam porta nunc a lobortis facilisis. Curabitur tincidunt mollis nisl, scelerisque fermentum diam ultrices et. Phasellus eros sapien, mollis nec sodales at, sodales sit amet ante.</Text>
      </NativeViewGestureHandler>
    </View>
  );
}
```

**Comment by cdlarson91** [comment]
> Hi! Could you try wrapping your selectable text views with `NativeViewGestureHandler`? I believe that it works correctly after this change:
> 
> ```js
> import { NativeViewGestureHandler } from 'react-native-gesture-handler';
> import * as React from 'react';
> import { Text, View, StyleSheet } from 'react-native';
> import { NavigationContainer } from '@react-navigation/native';
> import { createMaterialTopTabNavigator } from "@react-navigation/material-top-tabs";
> 
> const Tab = createMaterialTopTabNavigator();
> 
> export default function App() {
>     return (
>     <NavigationContainer>
>       <Tab.Navigator 
>         swipeEnabled={true}>
>         <Tab.Screen name="A" component={AScreen} />
>         <Tab.Screen name="B" component={BScreen} />
>       </Tab.Navigator>
>     </NavigationContainer>
>   );
> }
> 
> function AScreen() {
>   return (
>     <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
>       <NativeViewGestureHandler>
>         <Text selectable>Lorem ipsum dolor sit amet, consectetur adipiscing elit. In non est eget elit pretium elementum. Praesent at metus elementum, auctor turpis ac, tristique ex. Praesent malesuada aliquam lacus, nec vestibulum lorem venenatis sed. Nullam tincidunt tristique massa, quis molestie metus. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce arcu ex, pulvinar eu aliquet sed, facilisis et arcu. Proin sit amet mi est.</Text>
>       </NativeViewGestureHandler>
>     </View>
>   );
> }
> 
> function BScreen() {
>   return (
>     <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
>       <NativeViewGestureHandler>
>         <Text selectable>Sed leo lacus, bibendum cursus dignissim a, suscipit eget nisi. Aliquam vestibulum eros eget sem feugiat tincidunt placerat vel lorem. Pellentesque odio ligula, elementum ullamcorper urna at, cursus pharetra lectus. Aenean laoreet egestas sapien quis congue. Donec dictum, ex at congue faucibus, orci tortor pharetra libero, in laoreet velit enim eu velit. Aliquam porta nunc a lobortis facilisis. Curabitur tincidunt mollis nisl, scelerisque fermentum diam ultrices et. Phasellus eros sapien, mollis nec sodales at, sodales sit amet ante.</Text>
>       </NativeViewGestureHandler>
>     </View>
>   );
> }
> ```

That seems to have fixed the issue for me! Thanks!!

**Comment by pierroo** [comment]
I don't mean to expand this issue into a new one, but it felt very much related to mine; unfortunately this suggestion doesn't work for "touchable" in between swipes.
They still get triggered in spite of this wrapper "NativeViewGestureHandler".
Please let me know if I am polluting the initial issue, but if you happen to pass by @j-piasecki please ket me know if you know of another "hack" that could fix the touchable being triggered upon swiping between screens?
Don't know if that matters, but those touchables are inside a FlatList.

**Comment by j-piasecki** [comment]
@pierroo You could try changing the import of the Touchables from `react-native` to `react-native-gesture-handler`, however those might need a bit of tinkering to work in the same way. Most notably, they are composed of two views: `GestureHandlerButton` wrapping an `Animated.View`. It's likely that you would need to style both of them - `containerStyle` applies styling to the outer view - `GestureHandlerButton`, and `style` applies styling to the inner view - `Animated.View`.

**Comment by j-piasecki** [comment]
That is very interesting indeed. It should have a ripple effect, but I cannot think of any solution off the top of my head. Since that is (most probably) unrelated to this issue, could you make a new one?

**Comment by pierroo** [comment]
Definitely, that went beyond the initial topics; both your suggestion for OP and me worked, so thank you again so much for your help! :)
(btw, just in case anyone passes by, I imported "RectButton" from react-native-gesture-handler that does seem to have the ripple effect and also fix my issue, I am investigating if it brings out any new issue)

**Comment by j-piasecki** [comment]
Great! I will close this issue then. If you find any problems, please make a new one 😄.

**Comment by weishengkui** [comment]
> ```js
> NativeViewGestureHandler
> ```

But it doesn't work for react-native-webview. @j-piasecki @pierroo
